### PR TITLE
Move daemon into internal directory and to cmd/daemon/daemon.go

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

File: api/v1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -113,12 +113,27 @@ func (in *DpuOperatorConfigStatus) DeepCopy() *DpuOperatorConfigStatus {
 	return out
 }
 
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
```
Comment (wizhaoredhat) [inline api/v1/zz_generated.deepcopy.go:116]: Did you try building with these changes to deepcopy? It fails for me.

Comment (bn222) [reply api/v1/zz_generated.deepcopy.go:116]: We need to fix the discrepancy between your env and mine in a separate PR before we tackle this new PR.

Comment (bn222) [reply api/v1/zz_generated.deepcopy.go]: removing this change manually from the commit for now. We will bump to 1.22.3 early next week.

**Pull request merged**